packages/grafana-ui/src/components/DataSourceSettings/HttpProxySettings.tsx

```diff
@@ -34,36 +41,37 @@ export const HttpProxySettings = ({
             }
           />
         </InlineField>
-      </div>
-      <div className="gf-form-inline">
-        <InlineField label="Skip TLS Verify" labelWidth={LABEL_WIDTH} disabled={dataSourceConfig.readOnly}>
+      </Stack>
+      <InlineField label="Skip TLS Verify" labelWidth={LABEL_WIDTH} disabled={dataSourceConfig.readOnly}>
+        <InlineSwitch
+          id="http-settings-skip-tls-verify"
+          value={dataSourceConfig.jsonData.tlsSkipVerify || false}
+          onChange={(event) => onChange({ ...dataSourceConfig.jsonData, tlsSkipVerify: event!.currentTarget.checked })}
+        />
+      </InlineField>
+      {showForwardOAuthIdentityOption && (
+        <InlineField
+          label="Forward OAuth Identity"
+          tooltip="Forward the user's upstream OAuth identity to the data source (Their access token gets passed along)."
+          labelWidth={LABEL_WIDTH}
+          disabled={dataSourceConfig.readOnly}
+        >
           <InlineSwitch
-            id="http-settings-skip-tls-verify"
-            value={dataSourceConfig.jsonData.tlsSkipVerify || false}
+            id="http-settings-forward-oauth"
+            value={dataSourceConfig.jsonData.oauthPassThru || false}
             onChange={(event) =>
-              onChange({ ...dataSourceConfig.jsonData, tlsSkipVerify: event!.currentTarget.checked })
+              onChange({ ...dataSourceConfig.jsonData, oauthPassThru: event!.currentTarget.checked })
             }
           />
         </InlineField>
-      </div>
-      {showForwardOAuthIdentityOption && (
-        <div className="gf-form-inline">
-          <InlineField
-            label="Forward OAuth Identity"
-            tooltip="Forward the user's upstream OAuth identity to the data source (Their access token gets passed along)."
-            labelWidth={LABEL_WIDTH}
-            disabled={dataSourceConfig.readOnly}
-          >
-            <InlineSwitch
-              id="http-settings-forward-oauth"
-              value={dataSourceConfig.jsonData.oauthPassThru || false}
-              onChange={(event) =>
-                onChange({ ...dataSourceConfig.jsonData, oauthPassThru: event!.currentTarget.checked })
-              }
-            />
-          </InlineField>
-        </div>
       )}
-    </>
+    </div>
   );
 };
+
+const getGridLayout = (theme: GrafanaTheme2) =>
+  css({
+    display: 'grid',
+    gridTemplateColumns: 'auto',
+    gap: 0, // Inline field has a margin
+  });
```

Comment:
I think we should support this use case in the `Grid` component. Having the columns auto size based on the content.